src/sql/workbench/browser/ui/infoBox/infoBox.ts

```diff
@@ -98,14 +113,89 @@ export class InfoBox extends Disposable implements IThemable {
 		this.updateStyle();
 	}
 
+	public get links(): azdata.LinkArea[] {
+		return this._links;
+	}
+
+	public set links(v: azdata.LinkArea[]) {
+		this._links = v ?? [];
+	}
+
 	public get text(): string {
 		return this._text;
 	}
 
 	public set text(text: string) {
 		if (this._text !== text) {
 			this._text = text;
-			this._textElement.innerText = text;
+
+			DOM.clearNode(this._textElement);
+			this._linkListenersDisposableStore.clear();
+
+			for (let i = 0; i < this._links.length; i++) {
+				const placeholderIndex = text.indexOf(`{${i}}`);
+				if (placeholderIndex < 0) {
+					this._logService.warn(`Could not find placeholder text {${i}} in text ${text}`);
+					// Just continue on so we at least show the rest of the text if just one was missed or something
+					continue;
+				}
+
+				// First insert any text from the start of the current string fragment up to the placeholder
+				let curText = text.slice(0, placeholderIndex);
+				if (curText) {
+					const span = DOM.$('span');
+					span.innerText = text.slice(0, placeholderIndex);
+					this._textElement.appendChild(span);
+				}
+
+				// Now insert the link element
+				const link = this._links[i];
+				const linkElement = DOM.$('a', {
+					href: link.url === '' ? ' ' : link.url
```

Comment:
when I set href to '' the link looks like <a href></a> and it looks visited in the UI (check link 2)
![image](https://user-images.githubusercontent.com/6816294/160894458-9002db38-44da-4120-a735-323e04db5b8b.png)

this line of code fixes that.

Reply:
1. This is where comments are super useful since that isn't immediately obvious from looking at the code
2. Do you know why? Just changing it to something else without knowing why this is happening isn't great. I haven't dug much into how exactly electron determines link visitation but I'd start with the basic question of how this is determined and go from there.

Reply:
Using this issue #19007 to track it.